pjlib/src/pj/pool_caching.c

```diff
@@ -75,8 +75,12 @@ PJ_DEF(void) pj_caching_pool_init( pj_caching_pool *cp,
     cp->factory.create_pool = &cpool_create_pool;
     cp->factory.release_pool = &cpool_release_pool;
     cp->factory.dump_status = &cpool_dump_status;
-    cp->factory.on_block_alloc = &cpool_on_block_alloc;
-    cp->factory.on_block_free = &cpool_on_block_free;
+
+    /* Deprecated, these callbacks are only used for updating cp.used_size &
+     * cp.peak_used_size which are no longer used.
```

Comment:
They are still used in:
* unit tests such as `pjsip-perf.c` and `pjsip/src/test/test.c`
* `pjturn-srv/main.c`
Although mostly for logging purposes only.

Reply:
Ah right! Then I guess our options:
- change caching pool's mutex type to recursive mutex, but it may have some impact on performance (e.g: [here](https://stackoverflow.com/questions/187761/recursive-lock-mutex-vs-non-recursive-lock-mutex)), affected functions: `pj_pool_create/release()` (less impact because less frequently used?), `pj_pool_alloc()/reset()` (more impact, depends on initial & growing capacity). Perhaps avoid this approach as the fields are for logging purpose.
- leave it as is with possibility of data race, which may reduce accuracy of those fields, and trigger semi 'false alarm' on analyzing tools.
- use `pj_mutex_try_lock()`, when fails keep going without lock, assuming the failure is caused by having been locked by upstream/caller, e.g: `pj_pool_create/release()`
- ...?

Reply:
since the usage is mostly for logging and tests only, I suppose just comment/remove the code that use the deprecated fields?